server/api.js

```diff
@@ -9,29 +9,107 @@ router.get("/", (req, res) => {
 
 router.get("/clients", (req, res) => {
 	db.query("SELECT * from clients;")
-    .then((result) => res.json(result.rows))
-	.catch((e) => {
-		console.error(e);
-		res.sendStatus(400);
-	});
+		.then((result) => res.json(result.rows))
+		.catch((e) => {
+			res.status(400).send(e);
+		});
 });
 
-router.post("/clients", (req, res) => {
-
-	const newClientDateAdded = req.body.date_added;
-	const newClientName = req.body.name;
-	const newClientBikes = req.body.bikes_needed;
-	const newClientPhoneNumber = req.body.phone_number;
+const saveUser = async (element) => {
+	return new Promise((resolve, reject) => {
+		const newClientDateAdded = element.date_added;
+		if (!newClientDateAdded) {
+			return reject(
+				"The field date added is empty, it is require to have a date"
+			);
+		}
+		const newClientDateAddedIso = new Date(
+			Date.parse(newClientDateAdded)
+		).toISOString();
+		const newClientName = element.name;
+		if (!newClientName) {
+			return reject(
+				"The field name is empty and it is require to have client name"
+			);
+		}
+		const newClientBikes = element.bikes_needed;
+		if (!newClientBikes || isNaN(!newClientBikes)) {
```

Comment:
It should be isNaN(newClientBikes), isNaN(!newClientBikes) it will always return false